airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -0,0 +1,268 @@
+                    else insight.get("name")
+                )
+                insight_record[key] = insight.get("values")[0]["value"]
+                if not insight_record.get("date"):
+                    insight_record["date"] = insight.get("values")[0]["end_time"]
+            yield self.state_filter(insight_record)
+
+    def _params(self) -> Iterator[List]:
+        buffered_start_date = max(self._state.add(minutes=1), pendulum.now().subtract(days=self.buffer_days))
+        end_date = pendulum.now()
+
+        while buffered_start_date <= end_date:
+            params_list = []
+            for period, metrics in self.METRICS_BY_PERIOD.items():
+                params_list.append(
+                    {
+                        "metric": metrics,
+                        "period": [period],
+                        "since": buffered_start_date.to_datetime_string(),
+                        "until": buffered_start_date.add(days=self.days_increment).to_datetime_string(),
+                    }
+                )
+            yield params_list
+            buffered_start_date = buffered_start_date.add(days=self.days_increment)
+
+    def _get_insight_records(self, params):
+        return self._api.account.get_insights(params=params)._queue
+
+
+class MediaAPI(StreamAPI):
+    INVALID_CHILDREN_FIELDS = ["caption", "comments_count", "is_comment_enabled", "like_count", "children"]
+
+    def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
+        children_fields = list(set(fields) - set(self.INVALID_CHILDREN_FIELDS))
+        media = self._get_media({"limit": self.result_return_limit}, fields)
+        for record in media:
+            record_data = record.export_all_data()
+            if record_data.get("children"):
+                record_data["children"] = [
+                    self._get_single_record(child_record["id"], children_fields).export_all_data()
+                    for child_record in record.get("children")["data"]
+                ]
+            yield record_data
+
+    def _get_media(self, params, fields) -> Iterator[Any]:
+        """
+        This is necessary because the functions that call this endpoint return
+        a generator, whose calls need decorated with a backoff.
+        """
+        return self._api.account.get_media(params=params, fields=fields)
+
+    def _get_single_record(self, media_id, fields) -> IGMedia:
+        return IGMedia(media_id).api_get(fields=fields)
+
+
+class StoriesAPI(StreamAPI):
+    def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
+        media = self._get_stories({"limit": self.result_return_limit}, fields)
+        for record in media:
+            record_data = record.export_all_data()
+            yield record_data
+
+    def _get_stories(self, params, fields: list) -> Iterator[Any]:
+        """
+        This is necessary because the functions that call this endpoint return
+        a generator, whose calls need decorated with a backoff.
```

Comment:
I don't see a backoff implemented anwhere, is this a TODO?

Reply:
It is in separate PR, because I made it as separate issue:
https://github.com/airbytehq/airbyte/pull/2326